synthesizer/process/src/cost.rs

```diff
@@ -60,7 +60,14 @@ pub fn deployment_cost<N: Network>(deployment: &Deployment<N>) -> Result<(u64, (
 /// Returns the *minimum* cost in microcredits to publish the given execution (total cost, (storage cost, finalize cost)).
 pub fn execution_cost<N: Network>(process: &Process<N>, execution: &Execution<N>) -> Result<(u64, (u64, u64))> {
     // Compute the storage cost in microcredits.
-    let storage_cost = execution.size_in_bytes()?;
+    let mut storage_cost = execution.size_in_bytes()?;
+
+    // Compute a quadratic storage cost penalty if above the size penalty threshold.
+    if storage_cost > N::EXECUTION_STORAGE_PENALTY_THRESHOLD {
+        storage_cost = storage_cost
+            .saturating_pow(N::EXECUTION_STORAGE_FEE_DEGREE)
```

Comment:
Is there a reason why the penalty is applied to the entire `storage_cost` rather than just overage?

A jump in fee by 20,010 for 1 additional byte seems a little concerning

|$(storageCost ^{degree})/scalingFactor$|new storage_cost|
|-|-|
|$(5001 ^2)/1000$|25,010|
|$(5100 ^2)/1000$|26,010|
|$(6000 ^2)/1000$|36,000|
|$(32,000 ^2)/1000$|1,024,000|
|$(64,000 ^2)/1000$|4,096,000|
|$(128,000 ^2)/1000$|16,384,000|

Reply:
> Is there a reason why the penalty is applied to the entire storage_cost rather than just overage?

Namely because as transaction size grows, block size does as well. Large blocks with many large transactions require each node to fetch these transmissions to complete their view of the `dag` which causes stress on the BFT and syncing processes. 

While perhaps not the most mathematically beautiful, it prioritizes network health by ensuring penalties are applied to larger transactions starting early. It's likely that most functions on the network will have transaction sizes under 5kb, which is the reasoning for the initially linear pricing. 

Pricing information like the table you crafted above can be added to network documentation ensure this is explained.

Reply:
The modifier is now changed to 5000 - so this discontinuity no longer exists.

Reply:
|$(storageCost ^{degree})/scalingFactor$|new storage_cost|
|-|-|
|$(5001 ^2)/5000$|5002|
|$(5100 ^2)/5000$|5202|
|$(6000 ^2)/5000$|7200|
|$(32,000 ^2)/5000$|204,800|
|$(64,000 ^2)/5000$|819,200|
|$(128,000 ^2)/5000$|3,276,800|